Libraries/LibC/stdio.cpp

```diff
@@ -415,6 +417,20 @@ long FILE::tell()
     return lseek(m_fd, 0, SEEK_CUR);
 }
 
+void FILE::reopen(int fd, int mode)
+{
+    // Dr. POSIX says: "Failure to flush or close the file descriptor
+    //                  successfully shall be ignored"
+    // and so we ignore any failures these two might have.
+    flush();
+    close();
+
```

Comment:
I *think* we have to drop the buffer here (as in, `m_buffer.drop()`). `flush()` normally does that (or equivalent), but flushing may not have succeeded, and since we ignore the error, it's up to us to ensure we don't retain any buffered data from the old file.

Reply:
I don't think it's dropped when something fails while writing, but I didn't look deep enough into it, so I'll call `drop()` there